src/components/KTabs/KTabs.vue

```diff
@@ -97,12 +92,16 @@ const handleTabChange = (tab: string): void => {
 
 const getTabSlotName = (tabHash: string): string => tabHash.replace('#', '')
 
-const getAnchorTabindex = (tab: Tab): string => {
-  if (tab.disabled) {
-    return '-1'
+const tabComponent = (tab: Tab) => {
+  if (tab.to) {
+    if (typeof tab.to === 'string') {
+      return { tag: 'a', attributes: { href: tab.disabled ? undefined : tab.to } }
+    } else if (typeof tab.to === 'object') {
+      return { tag: 'router-link', attributes: { to: tab.disabled ? undefined : tab.to } }
+    }
   }
 
-  return typeof props.anchorTabindex === 'number' && props.anchorTabindex >= -1 && props.anchorTabindex <= 32767 ? String(props.anchorTabindex) : '0'
+  return { tag: 'button', attributes: { ...(tab.disabled && { disabled: true }) } }
```

Comment:
issue: this was previously a `div` and now you're returning a `button`

Reply:
Yes, we discussed this with @Justineo and decided this would not be considered a breaking change

Reply:
Let's have that discussion here -- changing the DOM structure technically _is_ a breaking change.

Reply:
That depends on the public API we are providing. I believe this should be considered an implementation detail.

Reply:
I think the key lies in how we manage user expectations. We should convey to users as clearly as possible that relying on the DOM structure is unsafe, serves as an escape hatch, and carries the risk of breaking in the future.

In my previous work, as the provider of the component library, we would even periodically scan the code of consumer applications to identify potentially risky usage patterns (such as deep selectors) and provide reports to the users, further clarifying the associated risks.

Reply:
If you look at our own internal implementations of Kongponents, using `KButton` as an example, there are hundreds of customizations directly tied to the DOM structure.

Reply:
Kongponents might not be perfectly abstracted, but the risks are internal. For Portal, I think setting clear customization boundaries is key—unlimited customization makes future updates a lot harder.

Reply:
> unlimited customization makes future update

I agree; however, I still am against trivial changes that aren't related to a specific feature or critical fix. We shouldn't just be changing things "because we want to"

Reply:
> We shouldn't just be changing things "because we want to"

I agree with this. However, I also recognize that the changes being made here are not arbitrary. The intention is to improve the project.

> Yes, we discussed this with @Justineo and decided this would not be considered a breaking change

As for my earlier suggestion: rendering `<a>` elements for tabs with `to` shouldn’t be considered a breaking change, as `to` wasn’t supported before. However, changing a `<div>` to a `<button>` does align with the best practices outlined in [WAI's examples](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#examples). How about keeping this change? @adamdehaven

Reply:
The change to `button` is arbitrary if the proper `role` (and other) attributes are in place on the existing `div`, but yes, does require adding additional things you "get for free" with a button.

I'm just attempting to stress we need to be very stringent with DOM changes as it could very easily break downstream projects. I'll unblock the change if this is how it's preferred to go forward on this PR, but subjective changes should be pushed back on overall.

Reply:
> subjective changes

To be clear, using a button here is more proper, I'm speaking generally.

Reply:
Changed the element back to div here.